src/pages/AppDetails/AppDetailsPage.tsx

```diff
@@ -69,16 +89,48 @@ class AppDetails extends React.Component<RouteComponentProps<AppId>, AppDetailsS
       });
   };
 
+  fetchTrafficData = () => {
+    const node: NodeParamsType = {
+      app: this.props.match.params.app,
+      namespace: { name: this.props.match.params.namespace },
+      nodeType: NodeType.APP,
+
+      // unneeded
+      workload: '',
+      service: '',
+      version: ''
+    };
+    const restParams = {
+      duration: '600s',
```

Comment:
"duration" is the correct term for this, but it is confusingly referred to as "rateInterval" below, in TrafficDetails.   I think you should rename the "rateInterval" property in TrafficDetails to "duration".  Also, I'd suggest not using the 600 constant in multiple places.  Perhaps define a single constant and use it here, and also below.

Note that "rateInterval" should be used when working with the actual rateInterval to be applied in a rate() calculation with Prometheus.

Although it's not a blocker, I wonder if this fetch of data should be done in the TrafficDetails component when the tab is selected, and not proactively here.

These comments also apply to the former Workload details.

Reply:
You are right about the constants. I fixed that.

And yes... Initially I wanted to keep the `TrafficDetails` component stateless. But it does have some state. So, I think it's completely valid to move the API calls to `TrafficDetails`. I'll add that to my TODO list.